packages/xchain-radix/__tests__/client.test.ts

```diff
@@ -0,0 +1,254 @@
+import { GatewayStatusResponse } from '@radixdlt/babylon-gateway-api-sdk'
+import { Balance, Fees, Network, Tx, TxParams, XChainClientParams } from '@xchainjs/xchain-client/src'
+import { Client } from '@xchainjs/xchain-radix/src'
+import { Asset, baseAmount } from '@xchainjs/xchain-util'
+import {
+  mockCommittedDetailsResponse,
+  mockEntityDetailsResponse,
+  mockStreamTransactionsResponse,
+  mockTransactionPreviewResponse,
+} from '../__mocks__/mocks'
+import { XrdAsset } from '../src/const'
+
+describe('RadixClient Test', () => {
+  const createDefaultRadixClient = (): Client => {
+    const phrase = 'rural bright ball negative already grass good grant nation screen model pizza'
+    const params: XChainClientParams = {
+      network: Network.Mainnet,
+      phrase: phrase,
+      feeBounds: { lower: 1, upper: 5 },
+    }
+    return new Client(params, 'Secp256k1')
+  }
+
+  it('Invalid phrase is thrown', async () => {
+    const phrase = 'rural bright ball negative already grass good grant nation screen model'
+    const params: XChainClientParams = {
+      network: Network.Mainnet,
+      phrase: phrase,
+    }
+    expect(() => new Client(params, 'Ed25519')).toThrowError('Invalid phrase')
+  })
+
+  it('client should be able to get address', async () => {
+    const radixClient = createDefaultRadixClient()
+    const address: string = await radixClient.getAddressAsync()
+    expect(address).toBe('account_rdx16xmah09yu9p9ynrmuc8z3a206n02tsmmkdvlmnx3cgu4s9r59wsxt2')
+  })
+
+  it('client with Ed25519 curve should be able to get address', async () => {
+    const phrase = 'rural bright ball negative already grass good grant nation screen model pizza'
+    const params: XChainClientParams = {
+      network: Network.Mainnet,
+      phrase: phrase,
+    }
+    const radixClient = new Client(params, 'Ed25519')
+    const address: string = await radixClient.getAddressAsync()
+    expect(address).toBe('account_rdx12xhd0hmqcmu72r3zdsdm6zexw33gjk5y2n73dyxju2cvpvlr8rzy87')
```

Comment:
Did we check that this aligns with the wallet?

Reply:
hmm what exactly should I check with regards to the wallet?

Reply:
As in that both the client and the wallet would derive the same account address from the same mnemonic.

Reply:
added a couple of unit tests to validate keys, taking references from here: https://github.com/radixdlt/sargon/tree/0ffac13ece645c500fe74d2f854186c6340b4cd7/fixtures/vector